torch/nn/modules/linear.py

```diff
@@ -178,4 +179,78 @@ def extra_repr(self) -> str:
             self.in1_features, self.in2_features, self.out_features, self.bias is not None
         )
 
+
+class LazyLinear(LazyModuleMixin, Linear):
+    r"""A :class:`torch.nn.Linear` module with lazy initialization.
+
+    This module will initialize the `weight` and `bias` parameters after
+    the first call to ``forward`` is done. After that the module will
+    become a :class:`torch.nn.Linear` dropping all the lazy initialization
+    logic.
+
+    Be sure to run "dummy" ``forward`` once to initialize all parameters in case
+    that you need to access the weights before the actual "real" ``forward`` pass.
+
+    Note that lazy modules cannot validate if the shape is correct during
+    deserialization.  Also note that the initial weights values may become different
+    from the original (non-lazy) module even if the random seed is manually
+    configured, as the order of initialization is different from the original
+    one; especially, ``module.cuda()`` may cause the initialization to run on
+    a GPU.
+    The default value of lazy parameters is ``UninitializedParameter()``.
+
+    Args:
+        out_features: size of each output sample
+        bias: If set to ``False``, the layer will not learn an additive bias.
+            Default: ``True``
+
+    Attributes:
```

Comment:
I worry about the docs duplication between this and the original Linear module. Is there a way to avoid the redundant documentation so that updates to the linear module don't have to be reflected here, too?

Reply:
I added the `nn.lazy.LazyModuleMixin` to the documentation and put a pointer to it in the linear docs